Libraries/DirectMLX.h

```diff
@@ -3470,9 +3475,41 @@ namespace dml
         return out;
     }
 
-    // 
-    // TODO: NonZeroCoordinates
-    // 
+    struct NonZeroCoordinatesOutputs
+    {
+        Expression count;
+        Expression coordinates;
+    };
+    inline NonZeroCoordinatesOutputs NonZeroCoordinates(Expression input)
+    {
+        detail::GraphBuilder* builder = input.Impl()->GetGraphBuilder();
+        TensorDesc inputTensor = input.Impl()->GetOutputDesc();
+        const auto& inputTensorSizes = inputTensor.sizes;
+        uint32_t dimensionCount = static_cast<uint32_t>(inputTensorSizes.size());
+
+        TensorDimensions outputSizes = {};
```

Comment:
>{}; [](http://example.com/codeflow?start=39&length=3)

`= {1}; ` would be simpler (no need for the push_back).

The count tensor is simply a scalar value, and there are no dimensionality requirements of it (except that it consist of at least 1 dimension, and any other extra dimensions but be 1's too).

```
        TensorDescParameter outputCountTensor =
        {
            "OutputCount",
            desc.OutputCountTensor,
            ParameterType::RequiredOutput,
            TensorDataTypeMask::Uint32,
            1,8,                        // min/max rank
            TensorIndex::OutputCount,   // self index
            TensorIndex::None,          // matches data type
            TensorIndex::None,         // matches rank
            TensorIndex::None,          // matches sizes
        };
```